src/importer.cpp

```diff
@@ -689,6 +689,28 @@ std::string addUnitsAvoidingNameClash(const ModelPtr &model, const UnitsPtr &uni
     return units->name();
 }
 
+void flattenUnitsTree(const ModelPtr &model, const UnitsPtr &u, size_t index)
+{
+    if (u->isImport()) {
+        auto importSource = u->importSource();
+        auto importingModel = importSource->model();
+        auto importedUnits = importingModel->units(u->importReference());
+        auto importedUnitsCopy = importedUnits->clone();
+        importedUnitsCopy->setName(u->name());
+        model->replaceUnits(index, importedUnitsCopy);
+        for (size_t unitIndex = 0; unitIndex < importedUnits->unitCount(); ++unitIndex) {
+            const std::string reference = importedUnits->unitAttributeReference(unitIndex);
```

Comment:
Might want to replace `const std::string` with `auto`.

Reply:
I quite like my `const std::string`.